src/linker/Linker.Steps/TrimUsingVisibleMembers.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class TrimUsingVisibleMembers : BaseStep
+	{
+		readonly string fileName;
+
+		public TrimUsingVisibleMembers (string fileName)
+		{
+			this.fileName = fileName;
+		}
+
+		protected override void Process ()
+		{
+			AssemblyDefinition assembly = TrimUsingEntryPoint.LoadAssemblyFile (fileName, Context);
+
+			AssemblyAction action = Context.Annotations.GetAction (assembly);
+			switch (action) {
+			case AssemblyAction.CopyUsed:
+				Annotations.Mark (assembly.MainModule, new DependencyInfo (DependencyKind.RootAssembly, assembly));
+				goto case AssemblyAction.Copy;
+			case AssemblyAction.Copy:
+				// Mark Step wil take care of marking whole assembly
+				return;
+			case AssemblyAction.Link:
+				break;
+			default:
+				Context.LogError ($"Trimming assembly '{assembly.Name}' cannot use action '{action}'", 1035);
+				return;
+			}
+
+			TypePreserve preserve = TypePreserve.All |
+				(Annotations.HasLinkerAttribute<InternalsVisibleToAttribute> (assembly) ? TypePreserve.AccessibilityVisibleOrInternal : TypePreserve.AccessibilityVisible);
```

Comment:
This will not work unless we teach the linker attributes about this attribute - in the `LinkerAttributesInformation` .ctor - there's an if/else over all recognized attributes - we would have to add it there.

This also probably means that we're missing a test case for this.

Reply:
Didn't go with the cache after all